flocker/node/script.py

```diff
@@ -50,14 +52,29 @@ class ZFSAgentOptions(Options):
          "The port on the control service to connect to.", int],
     ]
 
-    def parseArgs(self, hostname, host):
-        # Passing in the 'hostname' (really node identity) via command
-        # line is a hack.  See
-        # https://clusterhq.atlassian.net/browse/FLOC-1381 for solution.
-        self["hostname"] = unicode(hostname, "ascii")
+    def parseArgs(self, host):
         self["destination-host"] = unicode(host, "ascii")
 
 
+def _get_external_ip(host, port):
+    """
+    Get an external IP address for this node that can in theory connect to
+    the given host and port.
+
+    :param host: A host to connect to.
+    :param port: The port to connect to.
+
+    :return unicode: IP address of external interface on this node.
+    """
```

Comment:
It would be good to refer to an issue for replacing this with control service-side node/address association.